### ISSUE TITLE
Set height / width when rendering

### ISSUE THREAD

**Issue by nhaberl** [alanjuden/MvcReportViewer#37]
Hey, awesome work!
Would like to ask if it is possible to set the height / width to the renderer so that it fits to my html block?

In the ReportViewer there is a property to set it.

**Comment by alanjuden** [comment]
Hey there @nhaberl, which height / width is it that you're referring too? If you're referring to the html container element that holds your report data I'm sure you could size it using css. If you're asking to change the dimensions of your report, you'll have to control that from SSRS. When you're designing your report you can set the sizing and it just comes through that size from the SSRS API and that's what is displayed from my code. I hope this helps!

Alan

**Comment by nhaberl** [comment]
What I mean is when I have a report width a ration 1:1 and 100 x 100 inches ... how can I achieve that this report fits into my 200x200px box on my website? (Or 300x300 and so on)

There is a need to scale down and in ReportViewer control you can set the height and width as paramters to the report which is to render and then the control scales it so it fits to the report container.

**Comment by alanjuden** [comment]
@nhaberl, you can control the width through CSS as I suggested above. I have just set the width in this rule of my .css class:

.ReportViewerHeader

The width is forced to 100% and if you drop that off or override it then you can control it.

**Comment by nhaberl** [comment]
Yes that's right for the header but I want that a report which renders in the size of 1000x1000px by example fits in a div tag which is 100x100px ... so I need a way that I can request that report in 100x100px not 1000x1000px respectively.

No scrollbars and so on ... the content should fit into the target div.

**Comment by alanjuden** [comment]
@nhaberl, I believe I understand what you're saying but when I run the report on my test server and don't have the width specified in the .css file, the outside of the report viewer shrinks down to fit the size. Have you tried it without the .css rule I mentioned and it wasn't functioning for you as expected?

**Comment by nhaberl** [comment]
yes that's right ... but think the way around ... the rendered report is bigger than the viewer ... thewn scrollbars appear in the viewer and I would like to achieve that thé report scales down so it fits into the viewer

**Comment by alanjuden** [comment]
Any chance you'd be willing to share a screenshot (with your data blanked out or something) so that I could see what you're seeing?

**Comment by nhaberl** [comment]
![reportviewer - my asp net application](https://user-images.githubusercontent.com/3939814/30864383-b7114362-a2d3-11e7-8f9e-78342e94e80a.png)

**Comment by nhaberl** [comment]
the goal now would be that the report gets shrinked so it fits into the viewer in width and adopts the height.
the winning feature would be when the report centers automatically afterwards or we have a chnace for setting the alignment.

**Comment by alanjuden** [comment]
I understand what you're saying now. I don't have any way around this as SSRS controls the output html that I'm rendering.

This link is an article from Microsoft that explains the settings that I have access to pass through the SSRS api to change the settings of the report rendering: https://docs.microsoft.com/en-us/sql/reporting-services/html-device-information-settings

The only thing I could possibly think that may have any impact on the width would be the "Zoom" property but I have not tried that at all. It would be a total guessing game on your part on if that zoom would be consistent across screen sizes or anything.

**Comment by nhaberl** [comment]
So do you know maybe how the ReportViewer controls handles this ?

**Comment by nhaberl** [comment]
I had a short chat with the ReportViewer team and they told me that they are doing the scaling / zoom with translate:scale() (as thought :-)) ... so maybe you could add a feature where width and height of the target container is sent with the request and after rendering the report get scaled so it fits into the container.

Upside down it would be great if the target container increases the height if the report is bigger ... same as full width when setting the zoom level)

**Comment by nhaberl** [comment]
Let me know if you would like to have further information
